clang/include/clang/CConv/Constraints.h

```diff
@@ -528,7 +528,7 @@ class ConstraintsEnv {
 
 public:
   ConstraintsEnv() : consFreeKey(0), useChecked(true) { environment.clear(); }
-  EnvironmentMap &getVariables() { return environment; }
+  const EnvironmentMap &getVariables() const { return environment; }
```

Comment:
Making this `const` ensure the map will not be modified?

Reply:
Yes, the first `const`. The second `const` ensures the getter doesn't change any state itself.